flink-baseline/flink-canary/src/main/java/com/ness/flink/canary/pipeline/domain/KafkaConfigs.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2021-2023 Ness Digital Engineering
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ness.flink.canary.pipeline.domain;
+
+
+import java.io.Serializable;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+@Builder
+@Getter
+@Setter
+@EqualsAndHashCode
```

Comment:
Explain why do you need it.
Also I don't see public (no args) constructor, please have a read
https://ness-nde.atlassian.net/wiki/spaces/RSH/pages/3582296067/Code+style+standards#Best-practice

Reply:
Builder is used in order to create the KafkaConfig object to be sent into the Flink stream (look in KafkaConfigsGenerator). Getter and setter are not used. I'm not sure what EqualsAndHashCode does. I will read about it first then decide if it is needed.

Reply:
Builder is used in KafkaConfigsGenerator to create a KafkaConfigs object to be sent down the Flink stream. Getter is used in HealthCheckFunction processElement method. It gets the KafkaConfigs fields and stores the fields into a Properties object. Setter is not used anywhere because there is no need to mutate the KafkaConfigs object. I don't know what EqualsAndHashCode does. I will read about it and then decide what to do.

I plan to keep Builder and Getter. I will remove Setter.

Reply:
Also added NoArgsConstructor and AllArgsConstructor.

Reply:
After reviewing EqualsAndHashCode, I don't think it is needed either. I will remove it.